src/modules/launcher/Plugins/Microsoft.Plugin.Calculator.UnitTest/ExtendedCalculatorParserTests.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using Microsoft.Plugin.Calculator;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using NUnit.Framework;
+using Wox.Plugin;
+
+namespace Microsoft.Plugin.Calculator.UnitTests
+{
+    [TestFixture]
+    public class ExtendedCalculatorParserTests
+    {
+        [TestCase(null)]
+        [TestCase("")]
+        [TestCase("  ")]
+        public void InputValid_ThrowError_WhenCalledNullOrEmpty(string input)
+        {
+            // Act
+            Assert.Catch<ArgumentNullException>(() => CalculateHelper.InputValid(input));
+        }
+
+        [TestCase(null)]
+        [TestCase("")]
+        [TestCase("  ")]
+        public void Interpret_ThrowError_WhenCalledNullOrEmpty(string input)
+        {
+            // Arrange
+            var engine = new CalculateEngine();
+
+            // Act
+            Assert.Catch<ArgumentNullException>(() => engine.Interpret(input));
+        }
+
+        [TestCase("42")]
+        [TestCase("test")]
+        public void Interpret_NoResult_WhenCalled(string input)
+        {
+            // Arrange
+            var engine = new CalculateEngine();
+
+            // Act
+            var result = engine.Interpret(input);
+
+            // Assert
+            Assert.AreEqual(default(CalculateResult), result);
+        }
+
+        [TestCase("2 * 2", 4D)]
+        [TestCase("-2 ^ 2", 4D)]
+        [TestCase("-(2 ^ 2)", -4D)]
+        [TestCase("2 * pi", 6.28318530717959D)]
+        [TestCase("round(2 * pi)", 6D)]
+        [TestCase("1 == 2", default(double))]
+        [TestCase("pi * ( sin ( cos ( 2)))", -1.26995475603563D)]
+        [TestCase("5.6/2",  2.8D)]
+        [TestCase("123 * 4.56", 560.88D)]
+        [TestCase("1 - 9.0 / 10", 0.1D)]
+        [TestCase("0.5 * ((2*-395.2)+198.2)", -296.1D)]
+        [TestCase("2+2.11", 4.11D)]
+        public void Interpret_NoErrors_WhenCalled(string input, decimal expectedResult)
+        {
+            // Arrange
+            var engine = new CalculateEngine();
+
+            // Act
+            var result = engine.Interpret(input, CultureInfo.InvariantCulture);
+
+            // Assert
+            Assert.IsNotNull(result);
+            Assert.AreEqual(expectedResult, result.Result);
+        }
+
+        [TestCase("0.100000000000000000000", 0.00776627963145224D)] // BUG: Because data structure
```

Comment:
@saahmedm Can we create bugs for these 3 issues if there aren't already.  You can reproduce this issue by copying the test case string into the Run and observing the result. 

![image](https://user-images.githubusercontent.com/56318517/92627856-d1558580-f280-11ea-8423-107380b7ba48.png)

Reply:
@ryanbodrug-microsoft in the main text I have the included the issues (under Note). These are closed, but there was an ticket that the whole engine would be replaced, so maybe add it there?

Reply:
> @ryanbodrug-microsoft in the main text I have the included the issues (under Note). These are closed, but there was an ticket that the whole engine would be replaced, so maybe add it there? I think this isn't blocking for this PR, so I mark this as Resolved

Yep definitely not blocking this PR.  I just wanted to make sure we had issues capturing this.  @crutkas.  I think some of the issues that we closed under a main tracking issue still happen.  We now round the result, but don't do anything to guard against the input to the mages engine.   Thanks for calling this out AFAIK.